### ISSUE TITLE
list: tabIndex is automatically inserted for mdListItem

### ISSUE THREAD

**Issue by kibertoad** [angular/material#9995]
**Actual Behavior**:
 -  `What is the issue? *` tabindex is set to "-1" for a **md-list-item** even when there is no need to do so.
 -  `What is the expected behavior?` For a system with manually defined tabIndexes for all elements that should have a11y support, there should be a way to avoid automatically generated indexes which mess up navigation order.

**CodePen** (or steps to reproduce the issue): * https://codepen.io/kibertoad/pen/VmvyMY
 -  `Details:` Element with id "listItem" should not have tabindex.

**Angular Versions**: *
 -  `Angular Version:` 1.5.9
 -  `Angular Material Version:` 1.1.1

**Additional Information**:
 -  `Browser Type: *` Any
 -  `Browser Version: *` Any
 -  `OS: *` Any
 -  `Stack Traces:`

----

We are fully prepared to provide a PR with a fix for this issue if there exists a solution to this issue that would be approved by maintainers to proceed. 

One way this could be solved would be to introduce an additional configuration service in which "setTabIndexAutomatically" property would be stored. 

Another way - introduce an html "data-noTabIndex" attribute which would mark elements that should not receive a -1 tabIndex from Angular Material.

**Comment by kibertoad** [comment]
Ping? Can we get an opinion on what the best solution would be?

**Comment by Splaktar** [comment]
I updated the [CodePen](https://codepen.io/Splaktar/pen/rNBypGR?editors=1010) to `1.1.20` and I was not able to reproduce this any longer.
